scripts/zkai_d128_residual_add_proof_input.py

```
@@ -0,0 +1,681 @@
+
+
+def require_signed_q8(value: Any, label: str) -> None:
+    require_signed_m31(value, label)
+    if not (-Q8_SEMANTIC_ABS_BOUND <= value <= Q8_SEMANTIC_ABS_BOUND):
+        raise D128ResidualAddInputError(f"{label} outside fixed-point q8 semantic bounds")
+
+
+def _load_source(path: pathlib.Path, label: str) -> dict[str, Any]:
+    try:
+        if path.is_symlink():
+            raise D128ResidualAddInputError(f"{label} evidence must not be a symlink: {path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)